lib/data/multipleBackendGateway.js

```diff
@@ -71,38 +21,6 @@ const multipleBackendGateway = {
                 callback(errors.InternalError);
             });
         }
-        // client is AWS SDK
-        if (client.clientType === 'aws_s3') {
-            const awsKey = _createAwsKey(keyContext.bucketName,
-               keyContext.objectKey, client.bucketMatch);
-            // TODO: if object to be encrypted, use encryption
-            // on AWS
-            return client.upload({
-                Bucket: client.awsBucketName,
-                Key: awsKey,
-                Body: stream,
-                Metadata: keyContext.metaHeaders,
-                ContentLength: size,
-            },
-               (err, data) => {
-                   if (err) {
-                       const log = createLogger(reqUids);
-                       log.error('err from data backend',
-                       { error: err, dataStoreName: client.dataStoreName });
-                       return callback(errors.InternalError);
-                   }
-                   const dataRetrievalInfo = {
-                       key: awsKey,
-                       dataStoreName: client.dataStoreName,
-                       dataStoreType: client.clientType,
-                       // because of encryption the ETag here could be
-                       // different from our metadata so let's store it
-                       // TODO: let AWS handle encryption
-                       dataStoreETag: data.ETag,
-                   };
-                   return callback(null, dataRetrievalInfo);
-               });
-        }
         return client.put(stream, size, keyContext,
             reqUids, (err, key) => {
```

Comment:
as discussed, please make sure the aws backend is returning what you expect.  here it is returning a whole object, not just the key